### PR TITLE
Add check_mode tests for shell/command/script and make skipped=True/changed=True mutually exclusive

### PR DESCRIPTION
**Pull request by s-hertel**
##### SUMMARY
Follow up to  https://github.com/ansible/ansible/pull/76353

##### ISSUE TYPE
- Test Pull Request

### CODE REVIEW

File: test/integration/targets/script/tasks/main.yml [hunk 1]
```diff
@@ -177,6 +177,7 @@
   assert:
     that:
       - _check_mode_test is changed
+      - _check_mode_test is skipped
```
Comment (bcoca) [inline test/integration/targets/script/tasks/main.yml:180]: this should not pass, only changed or skipped should be valid, not both

**Comment by s-hertel** [comment]
I went through the list of modules/actions with partial or no check mode support and found nothing other than command/shell (and script since #76353) returning 'changed: True' and 'skipped: True', so I fixed that and updated the tests. While fixing that for script, I realized that check mode should actually be partially supported, just like for command (via creates/removes). The large diff for script is just unindentation and looks better with `?w=1` added to the files url.

File: lib/ansible/plugins/action/script.py [hunk 2]
```diff
@@ -87,41 +87,47 @@ def run(self, tmp=None, task_vars=None):
             except AnsibleError as e:
                 raise AnsibleActionFail(to_native(e))
 
+            if self._play_context.check_mode:
```
Comment (bcoca) [inline lib/ansible/plugins/action/script.py]: use self._task instead (play_context is not updated per loop item)

Comment (s-hertel) [reply lib/ansible/plugins/action/script.py]: Fixed, good catch

File: lib/ansible/plugins/action/script.py [hunk 3]
```diff
@@ -87,41 +87,47 @@ def run(self, tmp=None, task_vars=None):
             except AnsibleError as e:
                 raise AnsibleActionFail(to_native(e))
 
+            if self._task.check_mode:
+                # check mode is supported if 'creates' or 'removes' are provided
+                # the task has already been skipped if a change would not occur
+                if self._task.args.get('creates') or self._task.args.get('removes'):
+                    result['changed'] = True
```
Comment (bcoca) [inline lib/ansible/plugins/action/script.py:94]: `raise AnsibleActionDone(result=result)`

File: lib/ansible/plugins/action/script.py [hunk 4]
```diff
@@ -87,41 +87,47 @@ def run(self, tmp=None, task_vars=None):
...
+                    return result
+                # If the script doesn't return changed in the result, it defaults to True,
+                # but since the script may override 'changed', just skip instead of guessing.
+                else:
+                    result['changed'] = False
+                    raise AnsibleActionSkip('Check mode is not supported for this task.')
```
Comment (bcoca) [inline lib/ansible/plugins/action/script.py]: raise AnsibleActionSkip('Check mode is not supported for this task.', result=result)

**Review by bcoca — APPROVED** [review]

**Comment by s-hertel** [comment]
rebuild_merge

**Pull request merged**